sensors/include/libsensors.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Sensors Library
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _LIBSENSORS_H_
+#define _LIBSENSORS_H_
+
+#include <stdint.h>
+#include <sys/ioctl.h>
+
+
+/* Available sensor types */
+
+#define SENSOR_TYPE_ACCEL (1 << 0)
+#define SENSOR_TYPE_BARO  (1 << 1)
+#define SENSOR_TYPE_GPS   (1 << 2)
+#define SENSOR_TYPE_GYRO  (1 << 3)
+#define SENSOR_TYPE_MAG   (1 << 4)
+#define SENSOR_TYPE_TEMP  (1 << 5)
+
+
+typedef unsigned int sensor_type_t;
+
+
+typedef struct {
+	uint32_t devId;
+	int16_t accelX; /* value [mm/s^2] */
+	int16_t accelY; /* value [mm/s^2] */
+	int16_t accelZ; /* value [mm/s^2] */
+	uint8_t reserved[2];
+} accel_data_t;
+
+
+typedef struct {
+	uint32_t devId;
+	uint32_t pressure; /* pressure in [Pa] */
+	uint32_t temp;     /* temperature value in Kelvin [K] */
+} baro_data_t;
+
+
+/* GPS position in WGS84 coordinates */
+typedef struct {
+	uint32_t devId;
+	int32_t lat;          /* latitude in 1E-7 degrees */
+	int32_t lon;          /* longitude in 1E-7 degrees */
+	int32_t hdop;         /* horizontal dilution of precision */
+	int32_t vdop;         /* vertical dilution of precision */
+	int32_t alt;          /* altitude in 1E-3 [m] (millimetres) above MSL */
+	int32_t altEllipsoid; /* altitude in 1E-3 [m] (millimetres) above Ellipsoid */
+	int32_t groundSpeed;  /* GPS ground speed, [mm/s] */
+	int32_t velNorth;     /* GPS North velocity, [mm/s] */
+	int32_t velEast;      /* GPS East velocity, [mm/s] */
+	int32_t velDown;      /* GPS Down velocity, [mm/s] */
+	uint32_t eph;         /* GPS horizontal position accuracy 1E-3 [m] (millimetres)*/
+	uint32_t epv;         /* GPS vertical position accuracy 1E-3 [m] (millimetres)*/
+	int32_t heading;      /* value in [mrad] */
+	int32_t headingOffs;  /* value in [mrad] */
+	int32_t headingAccur; /* value in [mrad] */
```

Comment:
IMHO 16 bit precision can be used.

None of these values exceeds 2π. With 10^3 precision we are far below `uint16_t` limit. Also only `headingOffs` may be negative so unsigned types can be used for heading and heading accuracy